### PR TITLE
Decode DateTimeOffset formats

### PR DESCRIPTION
**Pull request by MiYanni**
Fixes: https://github.com/Azure/autorest.csharp/issues/307

Follows a similar pattern to what was done for encoding DateTimeOffsets.

### CODE REVIEW

File: src/AutoRest.CSharp.V3/CodeGen/ClientWriter.cs [hunk 1]
```diff
@@ -47,9 +47,9 @@ public bool WriteClient(CodeWriter writer, ServiceClient operationGroup)
         private void WriteOperation(CodeWriter writer, ClientMethod operation, CSharpNamespace @namespace)
         {
             //TODO: Handle multiple responses
-            var schemaResponse = operation.ResponseType;
-            CSharpType? responseType = schemaResponse != null ? _typeFactory.CreateType(schemaResponse) : null;
-            CSharpType returnType = schemaResponse != null && responseType != null
+            ResponseBody? responseBody = operation.ResponseBodies.FirstOrDefault();
```
Comment (pakrym) [inline src/AutoRest.CSharp.V3/CodeGen/ClientWriter.cs]: If we don't use the array in generator let's not add an array to the model yet.

**Comment by pakrym** [comment]
You didn't reenable any tests here. Do we have any that cover affected scenarios?

File: src/AutoRest.CSharp.V3/CodeGen/ClientWriter.cs (hunk 1 above)
Comment (MiYanni) [reply src/AutoRest.CSharp.V3/CodeGen/ClientWriter.cs]: Fixed.

**Comment by MiYanni** [comment]
`GetComplexPrimitiveDateTimeRfc1123` was the test unignored by this PR.

File: src/AutoRest.CSharp.V3/CodeGen/ClientWriter.cs [hunk 2]
```diff
@@ -47,16 +47,16 @@ public bool WriteClient(CodeWriter writer, ServiceClient operationGroup)
...
             CSharpType? headerModelType = operation.Response.HeaderModel != null ? _typeFactory.CreateType(operation.Response.HeaderModel.Name) : null;
 
             CSharpType responseType = bodyType switch
             {
-                null when headerModelType == null => new CSharpType(typeof(Response)),
-                null => new CSharpType(typeof(ResponseWithHeaders<>), headerModelType),
+                null when headerModelType != null => new CSharpType(typeof(ResponseWithHeaders<>), headerModelType),
                 { } when headerModelType == null => new CSharpType(typeof(Response<>), bodyType),
                 { } => new CSharpType(typeof(ResponseWithHeaders<>), bodyType, headerModelType),
+                _ => new CSharpType(typeof(Response)),
```
Comment (MiYanni) [inline src/AutoRest.CSharp.V3/CodeGen/ClientWriter.cs]: Resharper was like, "You ain't covering all those cases." And I was like, "Hmm, really?" So, I just put the 'all null' case as the default.

**Pull request merged**